test/test_other.py

```diff
@@ -12642,8 +12642,8 @@ def test_default_to_cxx(self):
     self.assertContained("'string' file not found", err)
 
     # But it works if we pass and explicit language mode.
-    self.run_process([EMCC, '-c', 'cxxfoo.h', '-x', 'c++-header'])
-    self.run_process([EMCC, '-c', 'cxxfoo.h', '-x', 'c++'])
+    self.run_process([EMCC, '-x', 'c++-header', '-c', 'cxxfoo.h'])
+    self.run_process([EMCC, '-x', 'c++', '-c', 'cxxfoo.h'])
```

Comment:
Why change the test?

Reply:
Because emcc now passes these arguments though in the correct order:

Before it was re-ordering to put the source file at the end or `emcc -c cxxfoo.h -x c++-header' was becoming `emcc -x c++-header -c cxxfoo.h`.

Becuase `-x` (filetype) only effects the files that follow it this inadvertent / mistaken behaviour was fixing the command line hiding what should have been an error/warning from clang:

```
$ clang -c cxxfoo.h -x c++-header
clang: warning: '-x c++-header' after last input file has no effect [-Wunused-command-line-argument]
```

Reply:
In other words this was more of a bug in the test that emcc was ignored by emcc because it happened to re-order the command line flags.

Reply:
I see, thanks... I didn't realize `-x` affects files after it. I thought it was global...